### PR TITLE
Fix `tox --help` and `tox --help-ini` shows an error when run outside a directory with tox support files #1509 #1540

### LINKED ISSUE #1509: tox --help shows an error when run outside a directory with tox support files
[This is obviously minor, but figured I'd file it anyhow]

Running `tox --help` to just check usage options from no directory in particular (one that doesn't have a `tox.ini` etc.) shows an error:

```
⊙  tox --help                                                                                                                                                                                     julian@Air
ERROR: tox config file (either pyproject.toml, tox.ini, setup.cfg) not found
usage: tox [--version] [-h] [--help-ini] [-v] [-q] [--showconfig] [-l] [-a] [-c CONFIGFILE] [-e envlist] [--devenv ENVDIR] [--notest] [--sdistonly] [-p VAL] [-o] [--parallel--safe-build]
           [--installpkg PATH] [--develop] [-i URL] [--pre] [-r] [--result-json PATH] [--hashseed SEED] [--force-dep REQ] [--sitepackages] [--alwayscopy] [-s [val]] [--workdir PATH]
           [args [args ...]]

tox options
...
```

```
⊙  tox --version                                                                                                                                                                                  julian@Air
3.14.3 imported from /Users/julian/.local/share/virtualenvs/tox/site-packages/tox/__init__.py
```

### LINKED ISSUE #1540: tox --help-ini shows an error when run outside a directory with tox support files
similar to issue #1509  
Running tox --help-ini from no directory in particular (one that doesn't have a tox.ini etc.) shows an error:  
```
⊙  tox --help                                                                                                                                                                                     nkpro@Prime
ERROR: tox config file (either pyproject.toml, tox.ini, setup.cfg) not found
envdir          <path>   default: {toxworkdir}/{envname}
set venv directory -- be very careful when changing this as tox will remove this directory when recreating an environment
...
```

### PR DESCRIPTION
**Pull request by nkpro2000sr**
closes #1509, closes #1540 

## Thanks for contributing a pull request!

If you are contributing for the first time or provide a trivial fix don't worry too
much about the checklist - we will help you get started.

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [x] added/updated test(s)
- [ ] updated/extended the documentation
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`,`breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with "by :user:`<your username>`"
  * please use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files - by :user:`superuser`."
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -11,6 +11,18 @@ with advance notice in the **Deprecations** section of releases.
 
 .. towncrier release notes start
 
+v3.14.6 (2020-03-20)
+--------------------
```
Comment (gaborbernat) [inline docs/changelog.rst]: This file should not be touched, rather individual files into https://github.com/tox-dev/tox/blob/master/docs/changelog/README.rst

File: src/tox/config/__init__.py [hunk 2]
```diff
@@ -270,6 +270,10 @@ def parseconfig(args, plugins=()):
         pm.hook.tox_configure(config=config)  # post process config object
         break
     else:
+        if "--help" in args or "-h" in args:
```
Comment (gaborbernat) [inline src/tox/config/__init__.py]: shouldn't we extract this from the arg parse; I mean this solution will not work with ``tox -ah`` 🤔

Comment (nkpro2000sr) [reply src/tox/config/__init__.py]: ooh i forgot to pass through arg sparser..

File: docs/changelog.rst (hunk 1 above)
Comment (nkpro2000sr) [reply docs/changelog.rst]: done :+1:

File: src/tox/config/__init__.py (hunk 2 above)
Comment (nkpro2000sr) [reply src/tox/config/__init__.py]: solved. thank you :smiley:

**Comment by nkpro2000sr** [comment]
I have a doubt that my docs/changelog/1539.bugfix.rst is in correct format ?

**Comment by nkpro2000sr** [comment]
I don't know why it failed in pypy linux. It also failed with same error  
```
pypy linux - generate tox test environment
Bash exited with code '1'
```
alos in another PR #1537 
but in both [0 test failed](https://dev.azure.com/toxdev/tox/_build/results?buildId=2375&view=ms.vss-test-web.build-test-results-tab)

File: src/tox/config/__init__.py [hunk 3]
```diff
@@ -270,6 +270,8 @@ def parseconfig(args, plugins=()):
         pm.hook.tox_configure(config=config)  # post process config object
         break
     else:
+        if option.help or option.helpini:
+            return config
```
Comment (nkpro2000sr) [inline src/tox/config/__init__.py]: Exception for `--help` and `--help-ini` commands

File: tests/unit/test_z_cmdline.py [hunk 4]
```diff
@@ -84,20 +84,32 @@ def test_getvenv(self, initproj):
 def test_notoxini_help_still_works(initproj, cmd):
     initproj("example123-0.5", filedefs={"tests": {"test_hello.py": "def test_hello(): pass"}})
     result = cmd("-h")
-    msg = "ERROR: tox config file (either pyproject.toml, tox.ini, setup.cfg) not found\n"
-    assert result.err == msg
```
Comment (nkpro2000sr) [inline tests/unit/test_z_cmdline.py]: `--help` not going to show error messages anymore :smiley:

File: tox.ini [hunk 5]
```diff
@@ -38,6 +39,11 @@ commands = pytest \
            -n={env:PYTEST_XDIST_PROC_NR:auto} \
            {posargs:.}
 
+[testenv:pypy]
+deps =
+    pip >= 19.3.1
+    psutil <= 5.6.7
```
Comment (nkpro2000sr) [inline tox.ini]: psutil 5.7.0 is incompatible with pypy
```bash
$ pip install psutil
DEPRECATION: A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting psutil
  Using cached psutil-5.7.0.tar.gz (449 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/elcot/.pyenv/versions/pypy-5.7.1/envs/check/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-WpZ3yC/psutil/setup.py'"'"'; __file__='"'"'/tmp/pip-install-WpZ3yC/psutil/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-WpZ3yC/psutil/pip-egg-info
         cwd: /tmp/pip-install-WpZ3yC/psutil/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<module>", line 1, in <module>
      File "/tmp/pip-install-WpZ3yC/psutil/setup.py", line 46, in <module>
        from _compat import PY3  # NOQA
      File "/tmp/pip-install-WpZ3yC/psutil/psutil/_compat.py", line 120, in <module>
        "broken / incompatible Python implementation, see: "
    RuntimeError: broken / incompatible Python implementation, see: https://github.com/giampaolo/psutil/issues/1659
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by nkpro2000sr** [comment]
> This PR is on hold until we fix the master CI.

Finally, all test passed :smiley_cat:

File: tox.ini (hunk 5 above)
Comment (nkpro2000sr) [reply tox.ini]: [Repoted](https://github.com/giampaolo/psutil/issues/1720 "psutil") to psutil

**Comment by gaborbernat** [comment]
released via https://pypi.org/project/tox/3.14.6/

**Pull request merged**